pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1122,6 +1126,35 @@ public void writeBrokerDataOnZooKeeper(boolean force) {
         }
     }
 
+    /**
+     * sort bundles by load and select topK bundles for each broker.
+     * @return the number of bundles selected
+     */
+    private CompletableFuture<Integer> selectTopKBundle() {
+        CompletableFuture<Integer> completableFuture = new CompletableFuture<>();
+
+        executors.execute(() -> {
+            // make the bundle-data update and sorting executed in single thread
```

Comment:
Why do we run it on a single thread? Can we use `synchronized` here?

Reply:
Updated to use synchronized to protect bundleArr, thanks.